### DISCUSSION TITLE
Relationship b/w state and context

### DISCUSSION THREAD

**Discussion by ivanbaj** [statelyai/xstate#3415] [Q&A]
Is it possible to be in the same state with different context? For example can I return to idle state after a state that has updated the context?

**Comment by davidkpiano** [comment]
If I understand the question correctly, yes, you can be in the same state with a different context. Actions can `assign()` to update the context in the same state.

**Reply by ivanbaj** [reply]
Thank you. My confusion is/was about being in Idle state then transition to some flow with one or more states to fetch some data and ending back to Idle state. Or ending back to a state named SomeDataFetched to indicate the that the app is in a new state. I am having conceptual problem understanding what defines a state. My gut feel is the going back to Idle state makes more sense even though the context of the machine is in a new “state”.
